### PR TITLE
feat(object-observable): create object observable implementation

### LINKED ISSUE #20: create observable class
add description

### PR DESCRIPTION
**Pull request by kaczor6418**
In this PR, I have created an implementation of observable using [JS Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy)

closes #20

### CODE REVIEW

**Review by tosh00 — DISMISSED** [review]

File: .eslintrc.json [hunk 1]
```diff
@@ -175,5 +175,22 @@
...
     "accessor-pairs": "off",
     "@typescript-eslint/no-namespace": "off"
-  }
+  },
+  "overrides": [
+    // Disabled rules only for test files
+    {
+      "files": ["**/*.spec.ts"],
+      "rules": {
+        "no-unused-expressions": "off",
```
Comment (wslabosz) [inline .eslintrc.json:185]: There's a eslint plugin for mocha that could be worth looking into, as to not disable some rules https://www.npmjs.com/package/eslint-plugin-mocha

File: .github/workflows/check-pr.yml [hunk 2]
```diff
@@ -8,12 +8,17 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [18.x]
+        node-version: [18.0.0]
```
Comment (wslabosz) [inline .github/workflows/check-pr.yml]: Is that necessary, as right now we have the setup-node action?

File: test/mochaSetup.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import * as chai from 'chai';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
```
Comment (wslabosz) [inline test/mochaSetup.ts]: we could specify `ignoreFiles` in tsconfig.json (dunno which is better)

**Review by wslabosz — DISMISSED** [review]
Everything looks good, left some comments about eslint and tsconfig.

File: .eslintrc.json (hunk 1 above)
Comment (kaczor6418) [reply .eslintrc.json:185]: I think we are more interested in this plugin: https://www.npmjs.com/package/eslint-plugin-chai-friendly

File: .github/workflows/check-pr.yml (hunk 2 above)
Comment (kaczor6418) [reply .github/workflows/check-pr.yml]: fair point, I am going to remove this

File: test/mochaSetup.ts (hunk 3 above)
Comment (kaczor6418) [reply test/mochaSetup.ts]: :+1:

**Review by tosh00 — APPROVED** [review]

**Pull request merged**